source/common/stream_info/stream_info_impl.h

```diff
@@ -177,6 +178,10 @@ struct StreamInfoImpl : public StreamInfo {
 
   uint64_t bytesReceived() const override { return bytes_received_; }
 
+  absl::optional<StreamState> streamState() const override { return stream_state_; }
+
+  void setStreamState(StreamState stream_state) override { stream_state_ = stream_state; }
```

Comment:
the point of using a flag is to allow multiple values at once. I.E. here' you'd do `stream_state_ |= stream_state`. In fact, now you'd want to (again) rename this method to be (maybe) `addStreamState`, and add an additional method `removeStreamState` which would do `stream_state_ &= ~stream_state`.

Reply:
Discussed offline. "Adding HTTP state into this adds potential complexity that may be more error-prone and is not currently required." Let's keep it as a flag, but also not allow multiple values at once at this time. @kyessenov wdyt?

Reply:
Also noting that initially this change was intended to be able to print in access log by %STREAM_INFO%, so allowing multiple flags needs additional consideration in regard to access logs.

Reply:
That's fine with me.